src/core/DatetimeClasses.ts

```diff
@@ -0,0 +1,475 @@
+        }
+        return hourMinuteSecondString;
+    }
+}
+
+/**
+ * Months for HzLocalDate
+ * @internal
+ */
+enum Months {
+    January = 1,
+    February,
+    March,
+    April,
+    May,
+    June,
+    July,
+    August,
+    September,
+    October,
+    November,
+    December
+}
+
+/**
+ * ### Local date object
+ * * Represents date in year without timezone.
+ */
+export class HzLocalDate {
+    private static readonly dateRegex = /(-?\d+)-(\d\d)-(\d\d)/;
+
+    private constructor(readonly year: number, readonly month: number, readonly date: number) {
+        if (!Number.isInteger(year) || !Number.isInteger(month) || !Number.isInteger(date)) {
+            throw new RangeError('Illegal arguments given to HzLocalTime. All arguments must be integers.');
+        }
+        if (!(month >= 1 && month <= 12)) {
+            throw new RangeError('Month must be between 1-12');
+        }
+        if (!(year >= -999_999_999 && year <= 999_999_999)) {
+            throw new RangeError('Year must be between -999_999_999 - 999_999_999');
+        }
+
+        if (date < 1) {
+            throw new RangeError('Invalid date. Date cannot be less than 1');
+        }
+
+        if (date > 28) {
+            let maxDate = 31;
+            switch (month) {
+                case 2:
+                    maxDate = HzLocalDate.isLeapYear(this.year) ? 29 : 28;
+                    break;
+                case 4:
+                    maxDate = 30;
+                    break;
+                case 6:
+                    maxDate = 30;
+                    break;
+                case 9:
+                    maxDate = 30;
+                    break;
+                case 11:
+                    maxDate = 30;
+                    break;
+            }
+
+            if (date > maxDate) {
+                if (date == 29) {
+                    throw new RangeError(`Invalid date. February 29 as ${this.year} is not a leap year`);
+                }
+                throw new RangeError(`Invalid date. ${Months[this.month]} ${this.date}`);
+            }
+        }
+    }
+
+    /**
+     * @internal
+     * Static constructor that is used by constructor function
+     */
+    static _new(year: number, month: number, date: number) {
+        return new HzLocalDate(year, month, date);
+    }
+
+    /**
+     * @internal
+     * Implementation is taken from java IsoChronology.isLeapYear()
+     * @param year Year value
+     */
+    private static isLeapYear(year: number): boolean {
+        return (year & 3) == 0 && (year % 100 != 0 || year % 400 == 0);
+    }
+
+    /**
+     * Constructs a {@link HzLocalDate} object from string.
+     * @throws RangeError if string is not passed, or string format is wrong
+     * @param dateString String in the form of yyyy-mm-dd
+     */
+    static fromString(dateString: string): HzLocalDate {
+        if (typeof dateString !== 'string') {
+            throw new RangeError('String expected.');
```

Comment:
Would https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError be better? Similar for others